### PR TITLE
Initial Menu UI with 

### PR DESCRIPTION
**Pull request by Manghi**
Groundwork for the navigational menus were the primary focus of this request. Towards the end I started creeping into video/graphics. More focused effort towards modularization too.

General
- Logging support is added. Limit each message to a debug level for cleanliness.
- Explicitly using `rust-sdl2` v.0.25.0 to avoid compat conflicts with other versions
- Corrected panning (though this needs to be reworked again lol)
- I tried to write the menu code so that it was easily extendible in the future for draw() and update()  [went through a few revs of this hehe]

**New** Menu.rs
- Each menu view holds a container of items to be listed
- Current menus are Start Screen, Options, and Video/Audio/Gameplay settings

**New** Video.rs
- Fullscreen support
- Detect supported resolutions

**New** Utils.rs
- ::Graphics:: Common fn to draw text on screen for a given context

### CODE REVIEW

File: src/video.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            modes: Vec::new(),
+        }
+    }
+
+    pub fn add_mode(&mut self, new_mode: DisplayMode) {
+            self.modes.push(new_mode);
+    }
+
+    pub fn print_supported_modes(&self) {
+        if log_enabled!(LogLevel::Info) {
```
Comment (AaronM04) [inline src/video.rs]: Why check for `log_enabled!`? Why not just print `info!` lines? I'm assuming if the log is below the Info level, you wouldn't see anything from `info!`, right?

Comment (Manghi) [inline src/video.rs]: You're right, that is completely useless. I pulled it in from the example I saw and didn't think twice about it. I'll remove it in the next PR :smile:

Comment (AaronM04) [inline src/video.rs]: @Manghi It's not too late! Just push a fix to the `interface` branch in your repo. 👍

Comment (Manghi) [inline src/video.rs]: Clean up is live! Thanks for waiting

**Comment by AaronM04** [comment]
I noticed something: if you click on the menu, it draws cells in the grid when you finally start the game. It looks like mouse clicks are handled the same as when the game is started. Also, I hear the fan running, like a lot of stuff is happening, even though it's just a menu.

**Pull request merged**